lib/stencil-pull.utils.js

```diff
@@ -1,54 +1,79 @@
-const { fetchFile } = require('./utils/networkUtils');
+const fs = require('fs');
+const _ = require('lodash');
 
 const utils = {};
 
 module.exports = utils;
 
-utils.selectActiveTheme = (options, callback) => {
-    const [activeTheme] = options.themes.filter(theme => theme.is_active).map(theme => theme.uuid);
+utils.getChannelActiveTheme = async options => {
+    const { config: { accessToken }, apiHost, storeHash, channelId } = options;
 
-    callback(null, Object.assign({}, options, { activeTheme }));
+    const activeTheme = await themeApiClient.getChannelActiveTheme({ accessToken, apiHost, storeHash, channelId });
+
+    console.log('ok'.green + ` -- Fetched theme details for channel ${channelId}`);
+
+    return { ...options, activeTheme };
 };
 
-utils.startThemeDownloadJob = async options => {
-    const { config: { accessToken }, apiHost, activeTheme, storeHash } = options;
+utils.getThemeConfiguration = async options => {
+    const { config: { accessToken }, apiHost, storeHash, activeTheme, saved } = options;
+
+    const themeId = activeTheme.active_theme_uuid;
+
+    const configurationId = saved ? activeTheme.saved_theme_configuration_uuid
+        : activeTheme.active_theme_configuration_uuid;
 
-    const { jobId } = await themeApiClient.downloadTheme({
-        accessToken: accessToken,
-        themeId: activeTheme,
-        apiHost,
-        storeHash,
-    });
+    const remoteThemeConfiguration = await themeApiClient.getThemeConfiguration({ accessToken, apiHost, storeHash,
+                                                                                  themeId, configurationId });
 
-    return {
-        ...options,
-        jobId,
-    };
+    console.log('ok'.green + ` -- Fetched ${saved ? 'saved' : 'active'} configuration`);
+
+    return { ...options, remoteThemeConfiguration };
 };
 
-utils.downloadThemeConfig = async options => {
-    const { path: tempThemePath, cleanup } = await tmp.file();
+utils.mergeThemeConfiguration = async options => {
+    const { remoteThemeConfiguration } = options;
 
-    try {
-        await fetchFile(options.downloadUrl, tempThemePath);
-    } catch (err) {
-        throw new Error(`Unable to download theme config from ${options.downloadUrl}: ${err.message}`);
-    }
+    let rawConfig = fs.readFileSync('config.json');
+    let parsedConfig = JSON.parse(rawConfig);
+    let diffDetected = false;
 
-    console.log('ok'.green + ' -- Theme files downloaded');
-    console.log('ok'.green + ' -- Extracting theme config');
+    // For any keys the remote configuration has in common with the local configuration,
+    // overwrite the local configuration if the remote configuration differs
+    for (const [key, value] of Object.entries(remoteThemeConfiguration.settings)) {
+        if (key in parsedConfig.settings) {
+            // Check for different types, and throw an error if they are found
+            if (typeof parsedConfig.settings[key] !== typeof value) {
```

Comment:
What the user should do if the types were changed (e.g. `string[]` to `string` or `string` to `number`) and the user can't update the local config?

Reply:
Yes, this probably means a mistake was made - so I want to throw an error here to let the user know for manual intervention.